oauth2_http/java/com/google/auth/oauth2/AppEngineCredentials.java

```diff
@@ -79,19 +80,25 @@
   private transient Method getSignature;
   private transient String account;
 
-  AppEngineCredentials(Collection<String> scopes) throws IOException {
+  AppEngineCredentials(Collection<String> scopes, Collection<String> defaultScopes)
+      throws IOException {
     this.scopes = scopes == null ? ImmutableSet.<String>of() : ImmutableList.copyOf(scopes);
-    this.scopesRequired = this.scopes.isEmpty();
+    this.defaultScopes =
```

Comment:
implementation nit: wouldn't it be better to not save `defaultScopes` in the constructor, and instead set `scopes` property to `defaultScopes` when `scopes` is empty?

```java
if (scopes == null || scopes.isEmpty()) {
    this.scopes =
        defaultScopes == null ? ImmutableSet.<String>of() : ImmutableList.copyOf(defaultScopes);
} else {
    this.scopes = mmutableList.copyOf(scopes);
}
```

Reply:
yes this is much better. Just updated the code. Thanks!